kombu/async/timer.py

```diff
@@ -41,7 +41,8 @@ def to_timestamp(d, default_timezone=utc, time=monotonic):
         if d.tzinfo is None:
             d = d.replace(tzinfo=default_timezone)
         diff = _time() - time()
-        return max((d - EPOCH).total_seconds() - diff, 0)
+        epoch = datetime.utcfromtimestamp(0).replace(tzinfo=default_timezone)
```

Comment:
I don't believe that the timezone replacement here is proper. This won't convert the time to the new timezone, but simply replace the internal timezone.

So 9 AM UTC will "become" 9 AM JST.

Likewise, the epoch here might be transformed into "epoch + 9 hours". 

what you can go for instead is `default_timezone.normalize(datetime.utcfromtimestamp(0)`